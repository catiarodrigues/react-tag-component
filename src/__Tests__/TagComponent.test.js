import React from "react";
import { shallow, mount } from "enzyme";

import TagComponent from "../components/TagComponent";

describe("<TagComponent /> tests", () => {
  it("renders without errors and match snapshot", () => {
    const wrapper = shallow(<TagComponent />);

    expect(wrapper).toMatchSnapshot();
  });

  
  it("renders no tags on start", () => {
    const wrapper = mount(<TagComponent />);
    
    expect(wrapper.find("span")).toHaveLength(0);
  });
  
  describe("should add and remove tags", () => {
    const wrapper = mount(<TagComponent />);

    it("add tag", () => {
      wrapper.find("input").instance().value = "Sample";
      expect(wrapper.find("input").instance().value).toEqual("Sample");
      
      wrapper.find("input").simulate('keydown', { keyCode: 13 });
      expect(wrapper.find("span")).toHaveLength(1);
  
      expect(wrapper.find("span").last().text()).toEqual("Sample");
    })
    
    it("remove tag", () => {
      wrapper.find("button").simulate('click')
      expect(wrapper.find("span")).toHaveLength(0)
    })  
  })

});