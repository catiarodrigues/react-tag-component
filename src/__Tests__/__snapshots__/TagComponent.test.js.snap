// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagComponent /> tests renders without errors and match snapshot 1`] = `
<Fragment>
  <div
    className="field"
  >
    <label
      className="label"
    >
      Tags:
    </label>
    <input
      className="input"
      onKeyDown={[Function]}
      placeholder="your tag"
      type="text"
    />
    <p
      className="help"
    >
      Add a tag by pressing enter
    </p>
  </div>
  <div
    className="tags"
  />
</Fragment>
`;
